plugin/harness/execer.go

```diff
@@ -74,36 +80,37 @@ func (e *Execer) Exec(ctx context.Context) error {
 
 	// execute the plugin. the execution logic differs
 	// based on programming language.
-	if module := out.Run.Go.Module; module != "" {
-		// if the plugin is a Go module
-
-		slog.FromContext(ctx).
-			Debug("go build", slog.String("module", module))
-
-		// compile the code
-		binpath := filepath.Join(e.Source, "step.exe")
-		cmd := exec.Command("go", "build", "-o", binpath, module)
-		cmd.Env = e.Environ
-		cmd.Dir = e.Source
-		cmd.Stderr = e.Stderr
-		cmd.Stdout = e.Stdout
-		if err := cmd.Run(); err != nil {
+	if source := out.Run.Binary.Source; source != "" {
```

Comment:
This seems to support cases where the source of the binary is an external URL to be downloaded. However, IMHO, we should support paths to the local filesystem and the well-known plugin binaries can be already injected in the builder VM (via the packer image creation process). Most users should just need the default value (and should be able to override as needed to specify URL).  Is there a way to do that already? Maybe `file://path/to/local/file` can be used but we should keep it extensible by supporting different kinds of sources (local/remote). Example:

plugin.yaml:

```
Run:
  Binary:
    Source: local | remote
    Location: /path/to/binary | http://hostname/url/of/the/binary
```

Reply:
cc: @bradrydzewski @shubham149

Reply:
We would not be caching the binaries on vm ami's since it is hard to update them. We can add it later if required. If source contains file:// at the beginning, we can pick it from local.

Reply:
I think we should find a way to update them easily if needed. We should avoid downloads as much as possible. Also, let's think about the workflow for SMP (on-prem) and we'll need to document how they are supposed to download/seed such plugin binaries.